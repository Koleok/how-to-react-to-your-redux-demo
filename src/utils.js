const getRandomInt = (min, max) =>
  Math.floor(Math.random() * (Math.floor(max) - Math.ceil(min))) +
  Math.ceil(min)

// take a line from each poem and string them together
export const createPoem = poems =>
  poems
    .map(({ lines, linecount }) => lines[getRandomInt(0, linecount)])
    .filter(x => x)
    .join('\n')
