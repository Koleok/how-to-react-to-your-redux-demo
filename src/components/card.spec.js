import React from 'react'
import Card from './card'
import { toTree } from '../test.utils'

it('should handle a poem object', () => {
  expect(toTree(<Card author="me" title="the poem lives" />)).toMatchSnapshot()
})
