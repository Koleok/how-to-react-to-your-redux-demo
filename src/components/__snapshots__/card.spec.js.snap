exports[`test should handle a poem object 1`] = `
<Styled(Col)
  md={3}
  xs={6}>
  <styled.div>
    <styled.div>
      <h4>
        <b>
          me
        </b>
      </h4>
      <p>
        the poem lives
      </p>
    </styled.div>
  </styled.div>
</Styled(Col)>
`;
